src/Chainweb/Miner/Config.hs

```diff
@@ -68,20 +73,34 @@ newtype MinerCount = MinerCount { _minerCount :: Natural }
 -- -------------------------------------------------------------------------- --
 -- Mining Config
 
-validateMinerConfig :: ConfigValidation MiningConfig []
-validateMinerConfig c = do
-    when (_nodeMiningEnabled nmc) $ tell
-        [ "In-node mining is enabled. This should only be used for testing"
-        , "In order to use in-node mining, mining-coordination must be enabled, too"
-        ]
-    when (_nodeMiningEnabled nmc && not (_coordinationEnabled cc))
-        $ throwError "In-node mining is enabled but mining coordination is disabled"
-    when (_nodeMiningEnabled nmc && view minerId (_nodeMiner nmc) == "")
-        $ throwError "In-node Mining is enabled but no miner id is configured"
+validateMinerConfig :: ChainwebVersion -> ConfigValidation MiningConfig []
+validateMinerConfig v c = do
+    when (_nodeMiningEnabled nmc) $ do
+        tell
+            [ "In-node mining is enabled. This should only be used for testing"
+            , "In order to use in-node mining, mining-coordination must be enabled, too"
+            ]
+        when (not (_coordinationEnabled cc))
+            $ throwError "In-node mining is enabled but mining coordination is disabled"
+        when (view minerId (_nodeMiner nmc) == "")
+            $ throwError "In-node Mining is enabled but no miner id is configured"
+
+    when (_coordinationEnabled cc && isProd) $ do
+        when (hostArch `notElem` supportedArchs) $ do
+            throwError $ mconcat
+                [ "Unsupported host architecture for mining on production networks: " <> sshow hostArch <> "."
+                , " Supported architectures are " <> sshow supportedArchs
+                ]
   where
     nmc = _miningInNode c
     cc = _miningCoordination c
 
+    -- This is a heuristic and we are rather a little too restrictiv. In the
+    -- future we may also consider uname -m and/or cpuinfo (including flags) here.
+    --
+    supportedArchs = [ "x86_64" ]
```

Comment:
This value needs to be double checked for the release binaries, i.e. the binaries and docker images produced by the CI cabal builds once CI builds for this branch are available.

Reply:
Can we do a blacklist instead? I think we want to be conservative here, seeing as this prevents a node from even starting

Reply:
I for white listing. This only affects miners on mainnet. And for those we need strict rules.